ingestion-beam/bin/generate-avro-schemas.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+# Generate a new archive with avro schemas from the mozilla-pipeline-schema repository
+
+cd "$(dirname "$0")/.." || exit
+
+if [ ! -x  "$(command -v jsonschema-transpiler)" ]; then
+    echo "jsonschema-transpiler is not installed"
+    echo "Run 'cargo install --git https://github.com/acmiyaguchi/jsonschema-transpiler.git --branch dev'"
+    exit 1
+fi
+
+if [[ ! -f schemas.tar.gz ]]; then
+    echo "Run 'bin/download-schemas'"
```

Comment:
I still have in my mind that we should be generating avro and bq schemas directly in mozilla-pipeline-schemas. It would make it very easy to inspect the source schemas via the normal GitHub UI, and it also would mean that the avro and bq schemas would get packaged up in the mps tarball rather than needing to pull in a separate tarball for avro schemas or to generate a tarball here.

Has a decision been made to run the schema transpiler elsewhere for the production pipeline?

Reply:
I opened an issue in mozilla-services/mozilla-pipeline-schemas#306, which includes a link to a branch where all the generated schemas can be browsed. The trouble is the increase in the number of dependencies. Adding the transpiler and python3 dependencies would make it even hard to contribute to the schema repo. I'm still mulling over the best way to keep it cross-platform and accessible.

@fbertsch and I have been discussing an Airflow DAG for generating the schema artifacts that also integrate probe information. Reading tarballs in GCS from this output might be another alternative to checking everything into mps.

Reply:
Thanks for starting that discussion. I've jumped in there with my thoughts on the benefits of checking in the generated avro and bq schemas.